src/components/PostDetail.server.jsx

```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { marked } from "marked";
+import getPosts from "../server/posts";
+
+export default async function PostDetail({ permalink }) {
+  const list = await getPosts();
```

Comment:
The server component for PostDetail is simple, it is just an async function

see we can just call the server util `getPosts()` without data API endpoint. super cool